Content.Server/GameObjects/Components/Nutrition/DrinkComponent.cs

```diff
@@ -1,148 +1,179 @@
-                return 0;
-            }
-            return Math.Max(1, (int)Math.Ceiling((_contents.CurrentVolume / _transferAmount).Float()));
-        }
+        [ViewVariables]
+        public bool Empty => _contents.CurrentVolume.Float() <= 0;
 
+        private AppearanceComponent _appearanceComponent;
+        private bool _opened = false;
+        private string _soundCollection;
+        private string _originalName;
 
         public override void ExposeData(ObjectSerializer serializer)
         {
             base.ExposeData(serializer);
-            serializer.DataField(ref _useSound, "use_sound", "/Audio/items/drink.ogg");
-            // E.g. cola can when done or clear bottle, whatever
-            // Currently this will enforce it has the same volume but this may change. - TODO: this should be implemented in a separate component
-            serializer.DataField(ref _despawnOnFinish, "despawn_empty", false);
-            serializer.DataField(ref _finishPrototype, "spawn_on_finish", null);
+            serializer.DataField(ref _useSound, "useSound", "/Audio/items/drink.ogg");
+            serializer.DataField(ref _defaultToOpened, "isOpen", false); //For things like cups of coffee.
+            serializer.DataField(ref _soundCollection, "openSounds","canOpenSounds");
         }
 
-        protected override void Startup()
+        public override void Initialize()
         {
-            base.Startup();
-            _contents = Owner.GetComponent<SolutionComponent>();
+            base.Initialize();
+            Owner.TryGetComponent(out _appearanceComponent);
+            Owner.TryGetComponent<SolutionComponent>(out _contents);
             _contents.Capabilities = SolutionCaps.PourIn
                                      | SolutionCaps.PourOut
                                      | SolutionCaps.Injectable;
-            _drinking = false;
-            Owner.TryGetComponent(out AppearanceComponent appearance);
-            _appearanceComponent = appearance;
-            _appearanceComponent?.SetData(SharedFoodComponent.FoodVisuals.MaxUses, MaxVolume.Float());
-            _updateAppearance();
+            _opened = _defaultToOpened;
+            _originalName = Owner.Name;
+            if (_opened)
+            {
+                UpdateName();
+            }
+
+            UpdateAppearance();
         }
 
-        private void _updateAppearance()
+        void ISolutionChange.SolutionChanged(SolutionChangeEventArgs eventArgs)
         {
-            _appearanceComponent?.SetData(SharedFoodComponent.FoodVisuals.Visual, _contents.CurrentVolume.Float());
+            UpdateName();
+            UpdateAppearance();
         }
 
-        bool IUse.UseEntity(UseEntityEventArgs eventArgs)
+        private void UpdateName()
         {
-            UseDrink(eventArgs.User);
+            if (Opened)
+            {
+                if (Empty)
+                {
+                    Owner.Name = $"{_originalName} {Loc.GetString("(empty)")}";
```

Comment:
Do we even need an `unopened` text like that? I would maybe be nicer if you just saw it on the sprite/examine and not change the name.

Reply:
It's on both!